### ISSUE TITLE
Improve source rehydration latency metric

### ISSUE THREAD

**Issue by RobinClowers** [MaterializeInc/materialize#26996]
### Feature request

 Currently this value reset to null during hydration, but that's not very ergonomic. Instead, never set it to null, and have a `hydrated_at` timestamp column that tells you if a source is currently hydrating.

**Comment by teskje** [comment]
The new `hydration_time` column in `mz_compute_hydration_statuses` will have the same behavior or not showing a hydration time while a dataflow is still hydrating. I think that makes intuitive sense, but more importantly the way compute introspection relations are implemented doesn't support remembering values across envd restarts.

Looking at the code, changing the behavior of `mz_source_statistics` as proposed above should be easy. But I think we want to keep `mz_source_statistics` consistent with `mz_compute_hydration_statuses` in this regard. And given that changing the latter is not easy, changing the former seems like the wrong step to take.

Both `mz_source_statistics` and `mz_compute_hydration_statuses` are (or will be) "retained metrics relations" that keep historical data for 30 days. So they do contain previous hydration times, they just need to be queried using `AS OF`/`UP TO`. 

```sql
materialize=> SUBSCRIBE (SELECT rehydration_latency FROM mz_internal.mz_source_statistics_with_history WHERE id = 'u13804') AS OF AT LEAST 0 UP TO 1720009353000;
 mz_timestamp  | mz_diff | rehydration_latency
---------------+---------+---------------------
 1717417845000 |       1 | 00:00:13.509
 1717664583933 |       1 |
 1717664583933 |      -1 | 00:00:13.509
 1717664643934 |      -1 |
 1717664643934 |       1 | 00:00:22.324
 1718270688255 |      -1 | 00:00:22.324
 1718270688255 |       1 | 00:00:28.854
 1718270936367 |       1 |
 1718270936367 |      -1 | 00:00:28.854
 1718270996367 |      -1 |
 1718270996367 |       1 | 00:00:39.8
 1718874258361 |       1 |
 1718874258361 |      -1 | 00:00:39.8
 1718874318360 |      -1 |
 1718874318360 |       1 | 00:00:26.993
 1719479341888 |      -1 | 00:00:26.993
 1719479341888 |       1 | 00:00:30.651
(17 rows)
```

We have discussed this offline a bit and the main blocker for the console to using this approach was that `SUBSCRIBE` is not supported over the HTTP endpoint. We agreed that it seems reasonable to change this and at least support `SUBSCRIBE ... UP TO ...` queries.

A benefit of using this approach is that yet get the hydration times over the last 30 days (so roughly the last 4 hydrations), rather than just the last one. I imagine this can be useful, for example to show trends.

**Comment by RobinClowers** [comment]
@teskje I agree consistency is important across these relations. I filed https://github.com/MaterializeInc/materialize/issues/27998 to track supporting bounded subscribes over http, and we can close this.
